components/src/main/java/dev/arkbuilders/components/filepicker/ArkFilePickerFragment.kt

```diff
@@ -386,39 +408,25 @@ open class ArkFilePickerFragment :
         private const val PATH_PART_PADDING = 4f
     }
 
-    private fun pinFile(file: Path) {
-        val roots = mCurrentRootsWithFavorites?.keys
-        val root = roots?.find { root -> file.startsWith(root) }
-        val favorites = mCurrentRootsWithFavorites?.get(root)?.flatten()
-
-        root?.let {
+    private fun observeDataChanged() {
+        viewModel.getObservablePinFileResult().observe(this) { pinResult ->
+            when(pinResult) {
+                PinFileResult.PIN_AS_ROOT ->
```

Comment:
You don't need to add `LiveData` and `PinFileResult` enum. Look at the `handleSideEffect` function.
For more information check [mvi-orbit](https://github.com/orbit-mvi/orbit-mvi)